e2e/Microsoft.Azure.Devices.E2ETests.NetStandard/ProvisioningTests.cs

```diff
@@ -0,0 +1,329 @@
+        [DataRow(nameof(ProvisioningTransportHandlerAmqp), nameof(SecurityClientTpm), null, TransportFallbackType.TcpOnly)]
+        [DataRow(nameof(ProvisioningTransportHandlerAmqp), nameof(SecurityClientX509), X509EnrollmentType.Individual, TransportFallbackType.TcpOnly)]
+        [DataRow(nameof(ProvisioningTransportHandlerAmqp), nameof(SecurityClientX509), X509EnrollmentType.Group, TransportFallbackType.TcpOnly)]
+        [DataRow(nameof(ProvisioningTransportHandlerAmqp), nameof(SecurityClientTpm), null, TransportFallbackType.WebSocketOnly)]
+        [DataRow(nameof(ProvisioningTransportHandlerAmqp), nameof(SecurityClientX509), X509EnrollmentType.Individual, TransportFallbackType.WebSocketOnly)]
+        [DataRow(nameof(ProvisioningTransportHandlerAmqp), nameof(SecurityClientX509), X509EnrollmentType.Group, TransportFallbackType.WebSocketOnly)]
+        [DataRow(nameof(ProvisioningTransportHandlerMqtt), nameof(SecurityClientTpm), null, TransportFallbackType.TcpOnly)]
+        [DataRow(nameof(ProvisioningTransportHandlerMqtt), nameof(SecurityClientX509), X509EnrollmentType.Individual, TransportFallbackType.TcpOnly)]
+        [DataRow(nameof(ProvisioningTransportHandlerMqtt), nameof(SecurityClientX509), X509EnrollmentType.Group, TransportFallbackType.TcpOnly)]
+        [DataRow(nameof(ProvisioningTransportHandlerMqtt), nameof(SecurityClientTpm), null, TransportFallbackType.WebSocketOnly)]
+        [DataRow(nameof(ProvisioningTransportHandlerMqtt), nameof(SecurityClientX509), X509EnrollmentType.Individual, TransportFallbackType.WebSocketOnly)]
+        [DataRow(nameof(ProvisioningTransportHandlerMqtt), nameof(SecurityClientX509), X509EnrollmentType.Group, TransportFallbackType.WebSocketOnly)]
+        public async Task ProvisioningDeviceClient_InvalidIdScope_Register_Fail(
+            string transportType,
+            string securityType,
+            X509EnrollmentType? x509EnrollmentType,
+            TransportFallbackType? transportFallback)
+        {
+            using (ProvisioningTransportHandler transport = CreateTransportHandlerFromName(transportType, transportFallback))
+            using (SecurityClient security = CreateSecurityClientFromName(securityType, x509EnrollmentType))
+            {
+                ProvisioningDeviceClient provClient = ProvisioningDeviceClient.Create(
+                    InvalidIDScope,
+                    security,
+                    transport);
+
+                var cts = new CancellationTokenSource(PassingTimeoutMiliseconds);
+
+                _log.WriteLine("ProvisioningClient RegisterAsync . . . ");
+                DeviceRegistrationResult result = await provClient.RegisterAsync(cts.Token);
+
+                _log.WriteLine($"{result.Status}");
+                _log.WriteLine($"ProvisioningClient AssignedHub: {result.AssignedHub}; DeviceID: {result.DeviceId}");
+
+                Assert.AreEqual(ProvisioningRegistrationStatusType.Assigned, result.Status);
```

Comment:
>Assigned [](start = 67, length = 8)

Shouldn't also this be Unassigned?

Reply:
The test is `ProvisioningDeviceClient_ValidRegistrationId_Register_Ok`: the expectation here is that after RegisterAsync completed, the device is provisioned.

Reply:
Note: could be that the code shifted and this was meant for another test. I've corrected the E2E tests - PTAL.